cfme/tests/intelligence/reports/test_validate_chargeback_report.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+from cfme.intelligence.reports.reports import CustomReport
+from utils.providers import setup_a_provider
+from utils.log import logger
+
+used_cpu_rate = 0.02  # MHz
+allocated_cpu_count = 4
+used_memory_rate = 0.02  # MB
+allocated_memory = 9514
+used_networkio_rate = 0.005  # KBps
+used_diskio_rate = 0.005  # KBps
+average_CPU_used_in_MHz = 2500
+average_memory_used_in_MB = 9514
+cpu_total = 4
+
+
+@pytest.fixture(scope="module")
+def setup_first_provider():
+    setup_a_provider(validate=True, check_existing=True)
+
+
+def calculate_cpu_used_cost():
+    cpu_used_cost = average_CPU_used_in_MHz * used_cpu_rate
+    return cpu_used_cost
+
+
+def calculate_memory_used_cost():
+    cpu_used_cost = average_memory_used_in_MB * used_memory_rate
+    return cpu_used_cost
+
+
+def calculate_networkio_used_cost():
+    cpu_used_cost = average_memory_used_in_MB * used_networkio_rate
+    return cpu_used_cost
+
+
+def _cleanup_report(report):
```

Comment:
it might be nice to go for a pattern where the function siguature is `_ensure_report_cleanup(request, report)`
and define a inner function that is decorated with `request.addfinalizer`